### PR TITLE
[django] expand tests to cover tastypie

### PR DESCRIPTION
**Pull request by dcramer**
This also corrects an issue with recursion behavior in transaction names, exhibited by standard tastypie URLs.

Fixes GH-940

### CODE REVIEW

**Comment by dcramer** [comment]
breaking django changes out into https://github.com/getsentry/raven-python/pull/960 since I have no idea why the tests dont pass on Travis

**Comment by dcramer** [comment]
@mitsuhiko @mattrobenolt one of you mind taking a quick pass on review here?

File: raven/contrib/django/resolver.py [hunk 1]
```diff
@@ -56,8 +56,8 @@ def _resolve(self, resolver, path, parents=None):
 
         if parents is None:
             parents = [resolver]
-        else:
-            parents.append(resolver)
+        elif resolver not in parents:
+            parents = parents + [resolver]
```
Comment (mitsuhiko) [inline raven/contrib/django/resolver.py]: is this here guaranteed to be a list?

Comment (dcramer) [inline raven/contrib/django/resolver.py]: @mitsuhiko yeah parents is only set by this function

Comment (mitsuhiko) [inline raven/contrib/django/resolver.py]: I thought parents is passed in?

Comment (dcramer) [inline raven/contrib/django/resolver.py]: only by itself (recursively), and only as a list

**Pull request merged**